### PR TITLE
mirror-nixos-branch.pl: drop i686 for > 23.11

### PR DESCRIPTION
**Pull request by vcunat**
For the motivation and discussions around that see
https://github.com/NixOS/nixpkgs/pull/266164

### CODE REVIEW

File: mirror-nixos-branch.pl [hunk 1]
```diff
@@ -225,7 +225,9 @@ sub fetch {
             downloadFile("nixos.iso_gnome.aarch64-linux");
             downloadFile("nixos.iso_gnome.x86_64-linux");
 
-            downloadFile("nixos.iso_minimal.i686-linux");
+            if ($channelName ~ /nixos-2[0123]/) { # i686 dropped for > 23.11
```
Comment (stigtsp) [inline mirror-nixos-branch.pl:228]:
```suggestion
            if ($channelName =~ /nixos-2[0123]/) { # i686 dropped for > 23.11
```

Comment (stigtsp) [reply mirror-nixos-branch.pl]: tbh, I'd prefer something like:

```perl
if ($channelName =~ m/^nixos-2[0-3]$/) {
```

but yolo :D

Comment (vcunat) [reply mirror-nixos-branch.pl]: Ah, I should've accepted it here instead of using git.

**Comment by vcunat** [comment]
Deployed 🤞🏽

**Comment by vcunat** [comment]
I guess the next step is to recheck that updated channels look OK, in a couple days.  Though there won't be any stable one > 23.11 for some months anyway, so such a "test" doesn't have good coverage either.

**Comment by mweinelt** [comment]
Warrants updating the platform support tiers.

**Comment by vcunat** [comment]
I don't see us (newly) breaking any promise here.  This is the only relevant part I suspect:
https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md#tier-3-1

**Comment by vcunat** [comment]
I don't see us (newly) breaking any promise here.  https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md

I guess we'll just need to add `pkgsi686Linux.stdenv` into channel blockers.

**Pull request merged**